src/main/java/domain/Rank.java

```diff
@@ -0,0 +1,56 @@
+package domain;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public enum Rank {
+	NOTHING(0, 0, false),
+	FIFTH(5000, 3, false),
+	FOURTH(50000, 4, false),
+	THIRD(150000, 5, false),
+	SECOND(30000000, 5, true),
+	FIRST(2000000000, 6, false);
+
+	private final int money;
+	private final int match;
+	private final boolean bonus;
+
+	Rank(int money, int match, boolean bonus) {
+		this.money = money;
+		this.match = match;
+		this.bonus = bonus;
+	}
+
+	public static Rank getRank(int target, boolean bonus) {
+		return Arrays.stream(Rank.values())
+			.filter(rank -> rank.match == target)
+			.filter(rank -> rank.bonus == bonus)
+			.findFirst()
+			.orElse(NOTHING);
+	}
+
+	public static Map<Rank, Integer> getMap() {
+		final Map<Rank, Integer> rankMap = new LinkedHashMap<>();
+		Arrays.stream(values())
+			.filter(rank -> !rank.equals(NOTHING))
+			.forEach(rank -> rankMap.put(rank, 0));
+		return rankMap;
+	}
+
+	public boolean isNothing() {
+		return this.equals(NOTHING);
+	}
```

Comment:
Enum 비교는 동일성 비교로 하면 더 좋습니다!
```suggestion
	public boolean isNothing() {
		return this == NOTHING;
	}
```

Reply:
동일성과 동등성 키워드로 자세히 공부해보겠습니다!